diff -upr orig/drivers/gpu/drm/drm_connector.c /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_connector.c
--- orig/drivers/gpu/drm/drm_connector.c	2022-07-06 21:45:27.722155652 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_connector.c	2022-07-06 20:47:04.930351526 +0800
@@ -74,7 +74,7 @@ struct drm_conn_prop_enum_list {
 /*
  * Connector and encoder types.
  */
-static struct drm_conn_prop_enum_list drm_connector_enum_list[] = {
+struct drm_conn_prop_enum_list drm_connector_enum_list[] = {
 	{ DRM_MODE_CONNECTOR_Unknown, "Unknown" },
 	{ DRM_MODE_CONNECTOR_VGA, "VGA" },
 	{ DRM_MODE_CONNECTOR_DVII, "DVI-I" },
@@ -97,6 +97,7 @@ static struct drm_conn_prop_enum_list dr
 	{ DRM_MODE_CONNECTOR_SPI, "SPI" },
 	{ DRM_MODE_CONNECTOR_USB, "USB" },
 };
+EXPORT_SYMBOL(drm_connector_enum_list);
 
 void drm_connector_ida_init(void)
 {
diff -upr orig/drivers/gpu/drm/drm_drv.c /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_drv.c
--- orig/drivers/gpu/drm/drm_drv.c	2022-07-06 21:45:27.723155651 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_drv.c	2022-07-06 20:47:04.946351526 +0800
@@ -52,8 +52,10 @@ MODULE_AUTHOR("Gareth Hughes, Leif Delga
 MODULE_DESCRIPTION("DRM shared core routines");
 MODULE_LICENSE("GPL and additional rights");
 
-static DEFINE_SPINLOCK(drm_minor_lock);
-static struct idr drm_minors_idr;
+DEFINE_SPINLOCK(drm_minor_lock);
+EXPORT_SYMBOL(drm_minor_lock);
+struct idr drm_minors_idr;
+EXPORT_SYMBOL(drm_minors_idr);
 
 /*
  * If the drm core fails to init for whatever reason,
@@ -64,7 +66,8 @@ static struct idr drm_minors_idr;
  */
 static bool drm_core_init_complete;
 
-static struct dentry *drm_debugfs_root;
+struct dentry *drm_debugfs_root;
+EXPORT_SYMBOL(drm_debugfs_root);
 
 DEFINE_STATIC_SRCU(drm_unplug_srcu);
 
@@ -105,6 +108,21 @@ static void drm_minor_alloc_release(stru
 
 	spin_lock_irqsave(&drm_minor_lock, flags);
 	idr_remove(&drm_minors_idr, minor->index);
+
+	spin_lock_irqsave(&drm_minor_lock, flags);
+	idr_remove(&drm_minors_idr, minor->index);
+
+	spin_lock_irqsave(&drm_minor_lock, flags);
+	idr_remove(&drm_minors_idr, minor->index);
+
+	spin_lock_irqsave(&drm_minor_lock, flags);
+	idr_remove(&drm_minors_idr, minor->index);
+
+	spin_lock_irqsave(&drm_minor_lock, flags);
+	idr_remove(&drm_minors_idr, minor->index);
+
+	spin_lock_irqsave(&drm_minor_lock, flags);
+	idr_remove(&drm_minors_idr, minor->index);
 	spin_unlock_irqrestore(&drm_minor_lock, flags);
 }
 
@@ -467,6 +485,7 @@ void drm_dev_unplug(struct drm_device *d
 	 * finished.
 	 */
 	dev->unplugged = true;
+	//atomic_set(&dev->unplugged,1);
 	synchronize_srcu(&drm_unplug_srcu);
 
 	drm_dev_unregister(dev);
@@ -863,22 +882,11 @@ static void remove_compat_control_link(s
  */
 int drm_dev_register(struct drm_device *dev, unsigned long flags)
 {
-	const struct drm_driver *driver = dev->driver;
+	struct drm_driver *driver = dev->driver;
 	int ret;
-
-	if (!driver->load)
-		drm_mode_config_validate(dev);
-
-	WARN_ON(!dev->managed.final_kfree);
-
-	if (drm_dev_needs_global_mutex(dev))
-		mutex_lock(&drm_global_mutex);
-
-	ret = drm_minor_register(dev, DRM_MINOR_RENDER);
-	if (ret)
-		goto err_minors;
-
+	
 	ret = drm_minor_register(dev, DRM_MINOR_PRIMARY);
+	
 	if (ret)
 		goto err_minors;
 
@@ -897,8 +905,8 @@ int drm_dev_register(struct drm_device *
 	if (drm_core_check_feature(dev, DRIVER_MODESET))
 		drm_modeset_register_all(dev);
 
-	DRM_INFO("Initialized %s %d.%d.%d %s for %s on minor %d\n",
-		 driver->name, driver->major, driver->minor,
+        DRM_INFO("orig Initialized %s %d.%d.%d %s for %s on minor %d\n",
+	driver->name, driver->major, driver->minor,
 		 driver->patchlevel, driver->date,
 		 dev->dev ? dev_name(dev->dev) : "virtual device",
 		 dev->primary->index);
@@ -1003,11 +1011,14 @@ static int drm_stub_open(struct inode *i
 	DRM_DEBUG("\n");
 
 	minor = drm_minor_acquire(iminor(inode));
-	if (IS_ERR(minor))
+	if (IS_ERR(minor)) {
+		printk("drm_stub_open err to acquire\n");
 		return PTR_ERR(minor);
+	}
 
 	new_fops = fops_get(minor->dev->driver->fops);
 	if (!new_fops) {
+		printk("drm_stub_open err to get fops\n");
 		err = -ENODEV;
 		goto out;
 	}
@@ -1061,6 +1072,7 @@ static int __init drm_core_init(void)
 
 	drm_core_init_complete = true;
 
+	DRM_DEBUG("orig size of drm_device  %lu drm_driver %lu \n", sizeof(struct drm_device) ,sizeof(struct drm_driver) );
 	DRM_DEBUG("Initialized\n");
 	return 0;
 
diff -upr orig/drivers/gpu/drm/drm_file.c /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_file.c
--- orig/drivers/gpu/drm/drm_file.c	2022-07-06 21:45:27.723155651 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_file.c	2022-07-06 20:47:04.956351525 +0800
@@ -55,6 +55,7 @@
 
 /* from BKL pushdown */
 DEFINE_MUTEX(drm_global_mutex);
+EXPORT_SYMBOL(drm_global_mutex);
 
 bool drm_dev_needs_global_mutex(struct drm_device *dev)
 {
diff -upr orig/drivers/gpu/drm/drm_modeset_lock.c /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_modeset_lock.c
--- orig/drivers/gpu/drm/drm_modeset_lock.c	2022-07-06 21:45:27.725155651 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_modeset_lock.c	2022-07-06 20:47:04.990351523 +0800
@@ -75,7 +75,8 @@
  * lists and lookup data structures.
  */
 
-static DEFINE_WW_CLASS(crtc_ww_class);
+DEFINE_WW_CLASS(crtc_ww_class);
+EXPORT_SYMBOL(crtc_ww_class);
 
 /**
  * drm_modeset_lock_all - take all modeset locks
diff -upr orig/drivers/gpu/drm/drm_sysfs.c /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_sysfs.c
--- orig/drivers/gpu/drm/drm_sysfs.c	2022-07-06 21:45:27.726155651 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_sysfs.c	2022-07-06 20:47:05.008351522 +0800
@@ -51,6 +51,7 @@ static struct device_type drm_sysfs_devi
 };
 
 struct class *drm_class;
+EXPORT_SYMBOL(drm_class);
 
 static char *drm_devnode(struct device *dev, umode_t *mode)
 {
diff -upr orig/drivers/gpu/drm/drm_trace.h /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_trace.h
--- orig/drivers/gpu/drm/drm_trace.h	2022-07-06 21:45:27.726155651 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/drm_trace.h	2022-07-06 20:47:05.009351522 +0800
@@ -70,5 +70,5 @@ TRACE_EVENT(drm_vblank_event_delivered,
 
 /* This part must be outside protection */
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH ../../drivers/gpu/drm
+#define TRACE_INCLUDE_PATH /home/media/jingxuan/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm
 #include <trace/define_trace.h>
diff -upr orig/drivers/gpu/drm/Makefile /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/Makefile
--- orig/drivers/gpu/drm/Makefile	2022-07-06 21:45:27.717155652 +0800
+++ /home/jingxuan/372_upload/dg1_source_run/applications.services.cloud.visualcloud.vcdp.kmd-backport/orig/drivers/gpu/drm/Makefile	2022-07-06 20:47:05.740351481 +0800
@@ -26,7 +26,7 @@ drm-$(CONFIG_DRM_LEGACY) += drm_agpsuppo
 drm-$(CONFIG_DRM_LIB_RANDOM) += lib/drm_random.o
 drm-$(CONFIG_COMPAT) += drm_ioc32.o
 drm-$(CONFIG_DRM_GEM_CMA_HELPER) += drm_gem_cma_helper.o
-drm-$(CONFIG_DRM_GEM_SHMEM_HELPER) += drm_gem_shmem_helper.o
+#drm-$(CONFIG_DRM_GEM_SHMEM_HELPER) += drm_gem_shmem_helper.o
 drm-$(CONFIG_DRM_PANEL) += drm_panel.o
 drm-$(CONFIG_OF) += drm_of.o
 drm-$(CONFIG_PCI) += drm_pci.o
@@ -62,71 +62,7 @@ obj-$(CONFIG_DRM)	+= drm.o
 obj-$(CONFIG_DRM_MIPI_DBI) += drm_mipi_dbi.o
 obj-$(CONFIG_DRM_MIPI_DSI) += drm_mipi_dsi.o
 obj-$(CONFIG_DRM_PANEL_ORIENTATION_QUIRKS) += drm_panel_orientation_quirks.o
-obj-y			+= arm/
-obj-$(CONFIG_DRM_TTM)	+= ttm/
-obj-$(CONFIG_DRM_SCHED)	+= scheduler/
-obj-$(CONFIG_DRM_TDFX)	+= tdfx/
-obj-$(CONFIG_DRM_R128)	+= r128/
-obj-$(CONFIG_DRM_RADEON)+= radeon/
-obj-$(CONFIG_DRM_AMDGPU)+= amd/amdgpu/
-obj-$(CONFIG_DRM_MGA)	+= mga/
-obj-$(CONFIG_DRM_I810)	+= i810/
-obj-$(CONFIG_DRM_I915)	+= i915/
-obj-$(CONFIG_DRM_KMB_DISPLAY)  += kmb/
-obj-$(CONFIG_DRM_MGAG200) += mgag200/
-obj-$(CONFIG_DRM_V3D)  += v3d/
-obj-$(CONFIG_DRM_VC4)  += vc4/
-obj-$(CONFIG_DRM_SIS)   += sis/
-obj-$(CONFIG_DRM_SAVAGE)+= savage/
-obj-$(CONFIG_DRM_VMWGFX)+= vmwgfx/
-obj-$(CONFIG_DRM_VIA)	+=via/
-obj-$(CONFIG_DRM_VGEM)	+= vgem/
-obj-$(CONFIG_DRM_VKMS)	+= vkms/
-obj-$(CONFIG_DRM_NOUVEAU) +=nouveau/
-obj-$(CONFIG_DRM_EXYNOS) +=exynos/
-obj-$(CONFIG_DRM_ROCKCHIP) +=rockchip/
-obj-$(CONFIG_DRM_GMA500) += gma500/
-obj-$(CONFIG_DRM_UDL) += udl/
-obj-$(CONFIG_DRM_AST) += ast/
-obj-$(CONFIG_DRM_ARMADA) += armada/
-obj-$(CONFIG_DRM_ATMEL_HLCDC)	+= atmel-hlcdc/
-obj-y			+= rcar-du/
-obj-$(CONFIG_DRM_SHMOBILE) +=shmobile/
-obj-y			+= omapdrm/
-obj-$(CONFIG_DRM_SUN4I) += sun4i/
-obj-y			+= tilcdc/
-obj-$(CONFIG_DRM_QXL) += qxl/
-obj-$(CONFIG_DRM_BOCHS) += bochs/
-obj-$(CONFIG_DRM_VIRTIO_GPU) += virtio/
-obj-$(CONFIG_DRM_MSM) += msm/
-obj-$(CONFIG_DRM_TEGRA) += tegra/
-obj-$(CONFIG_DRM_STM) += stm/
-obj-$(CONFIG_DRM_STI) += sti/
-obj-y 			+= imx/
-# obj-$(CONFIG_DRM_INGENIC) += ingenic/
-obj-$(CONFIG_DRM_MEDIATEK) += mediatek/
-obj-$(CONFIG_DRM_MESON)	+= meson/
-obj-y			+= i2c/
-obj-y			+= panel/
-obj-y			+= bridge/
-obj-$(CONFIG_DRM_FSL_DCU) += fsl-dcu/
-obj-$(CONFIG_DRM_ETNAVIV) += etnaviv/
-obj-y			+= hisilicon/
-obj-$(CONFIG_DRM_ZTE)	+= zte/
-obj-$(CONFIG_DRM_MXSFB)	+= mxsfb/
-obj-y			+= tiny/
-obj-$(CONFIG_DRM_PL111) += pl111/
-obj-$(CONFIG_DRM_TVE200) += tve200/
-obj-$(CONFIG_DRM_XEN) += xen/
-obj-$(CONFIG_DRM_VBOXVIDEO) += vboxvideo/
-# obj-$(CONFIG_DRM_LIMA)  += lima/
-# obj-$(CONFIG_DRM_PANFROST) += panfrost/
-obj-$(CONFIG_DRM_ASPEED_GFX) += aspeed/
-# obj-$(CONFIG_DRM_MCDE) += mcde/
-# obj-$(CONFIG_DRM_TIDSS) += tidss/
-# obj-y			+= xlnx/
-obj-y			+= gud/
-obj-$(CONFIG_DRM_HYPERV) += hyperv/
+
 
 # Enable shims for the RHEL DRM backport
 subdir-ccflags-y += -DRH_DRM_BACKPORT
